### DISCUSSION TITLE
[@types/three] - most of the tests don't test types, they test the library.

### DISCUSSION THREAD

**Discussion by epreston** [DefinitelyTyped/DefinitelyTyped#64118] [Issues with a @types package]
Question about testing for type definitions:

Very few of the tests for the "three" type definitions test the types.  Most of them are simple unit tests for the library itself.  

Is this correct ?

I would like to contribute but I would like to understand the expectations.  My understanding is that each test to should test the type definitions, expected types, return types.  It is not a unit test suite for the library.  

Only `three/test/core/core-eventDispatcher.ts` matches what I was expecting from type tests.  Everything else is just a unit test for the library itself.

[a comment by an automated account is not shown]

**Comment by joshuaellis** [comment]
> Question about testing for type definitions:
> 
> Very few of the tests for the "three" type definitions test the types. Most of them are simple unit tests for the library itself.
> 
> Is this correct ?
> 
> I would like to contribute but I would like to understand the expectations. My understanding is that each test to should test the type definitions, expected types, return types. It is not a unit test suite for the library.
> 
> Only `three/test/core/core-eventDispatcher.ts` matches what I was expecting from type tests. Everything else is just a unit test for the library itself.

You should open this issue in [`three-ts-types`](https://github.com/three-types/three-ts-types) where we can discuss it more.

**Comment by epreston** [comment]
Sure, I can raise the issue there as well.
